Adding 2024 recap blog post

kmartens27 opened the pull request:
This PR is to add the 2024 recap blog post. **Note that this is not intended to be merged prior to January 8th**

In addition to the reviewers already requested, I am also asking if @janfaracik can provide any feedback or review if possible.

There is also a sponsors logo image being put together that will be added to the blog post in the sponsors section, but it is not available just yet. It will be added well before the intended merge date.

If there is anything that you feel is missing or should be included, please feel free to share that here as well!

kmartens27 commented:
Link to deploy preview: https://deploy-preview-7761--jenkins-io-site-pr.netlify.app/blog/2025/01/08/jenkins-2024-recap/

timja commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,176 @@
+They discussed how upgrades and the constantly changing landscape of development influence the Jenkins project's direction and what it takes to manage a large project like Jenkins.
+Thanks to Techstrong TV for having Jenkins on and to the Jenkins representatives for sharing their insights and anecdotes about the project.
+
+[[modern-ui]]
```
It would be good to get a picture into this section reflecting some of the changes in this year

gounthar replied:
Would that help?
![image](https://github.com/user-attachments/assets/c7a53a9c-ab15-48bf-8032-1cbdac9a54d0)

timja replied:
ideally something that encompasses the actual changes I think

janfaracik replied:
![bento](https://github.com/user-attachments/assets/55c39085-6200-4765-9595-8b42dcdb6bb5)

How is this?

timja replied:
Looks great

kmartens27 replied:
Thanks very much @janfaracik for creating and sharing that, I'll be sure to add it into the post!

timja commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,176 @@
+Additional thanks to all those who help enhance the Jenkins UX/UI through their efforts to provide the best experience possible.
+Constantly working to enhance the Jenkins user experience, the folks working on UI/UX have a passion for making Jenkins more sleek and modern.
+This passion translates to providing Jenkins users with a more friendly and enjoyable experience when using Jenkins, whether they're long-time users or just getting started.
+They have refreshed Jenkins menus and pages, link:https://www.jenkins.io/blog/2024/12/02/pr-titles/[incorporated user generated requests], and implemented several changes to refine the UI such as overhauling the search function.
```
YUI is also gone which was a major effort

janfaracik replied:
Maybe a short mention of the [redesigned builds widget](https://github.com/jenkinsci/jenkins/pull/9148) and revamped [v3 of the Design Library](https://github.com/jenkinsci/design-library-plugin/pull/359).

kmartens27 replied:
Both sound like good ideas, I'll add them to the UI/UX section along with the links provided here. Thank very much @timja and @janfaracik

alyssat reviewed (approved):
LGTM. Thank you @kmartens27 for this PR

gounthar commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,176 @@
+
+image:/images/post-images/2025/01/08/2024-recap.png[Jenkins 2024 recap]
+
+As we start 2025, we wanted to take a look back at the previous year and highlight some of our favorite moments, biggest projects, and important changes that occurred over the last 366 days.
+Regardless of what form a contribution takes, the Jenkins project consists of all of them and without them, we would not be what we are today.
+From every pull request, to the grants and donations that help Jenkins keep up with the needs of the Jenkins community, we offer our deepest thanks and appreciation.
```
```suggestion
As we start 2025, we wanted to look back at the previous year and highlight some of our favorite moments, biggest projects, and important changes over the last 366 days.
Regardless of what form a contribution takes, the Jenkins project consists of all of them and without them, we would not be what we are today.
From every pull request to the grants and donations that help Jenkins keep up with the needs of the Jenkins community, we offer our deepest thanks and appreciation.
```

gounthar commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,176 @@
+image:/images/post-images/2025/01/08/infrastructure.png[Infrastructure Update]
+
+Thanks to the efforts of the Jenkins infrastructure team, the Jenkins update center has been migrated to a new mirror-based architecture, which is 10x more cost-effective and is link:https://www.jenkins.io/blog/2024/11/16/new-update-center/[now in production].
+Over several months, Jenkins Infra performed multiple brownouts to test the services and ensure that there would be no issue with the migration.
+The tests were successful and they were able to detect issues that would have caused major headaches for users.
+Thankfully, these issues were resolved and while there were a few bumps post migration, the update center has been doing very well over the last few weeks.
```
```suggestion
Over several months, Jenkins Infra performed multiple brownouts to test the services and ensure that the migration would be smooth.
The tests were successful and they were able to detect issues that would have caused major headaches for users.
Thankfully, these issues were resolved and while there were a few bumps post-migration, the update center has been doing very well over the last few weeks.
```

gounthar commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,176 @@
+The tests were successful and they were able to detect issues that would have caused major headaches for users.
+Thankfully, these issues were resolved and while there were a few bumps post migration, the update center has been doing very well over the last few weeks.
+
+Throughout the year, the Jenkins infrastructure team has been working on several areas to reduce operating costs all around.
+This has taken the form of converting the spending and usage in some areas to others, migrating the update center, and finding new ways to alleviate the spending, all while increasing the overall Jenkins job count.
+Their efforts may be unseen at times, but they are always felt throughout the project.
```
```suggestion
Throughout the year, the Jenkins infrastructure team has been working on several areas to reduce operating costs all around.
This has taken the form of converting the spending and usage in some areas to others, migrating the update center, and finding new ways to alleviate the spending, all while increasing the overall Jenkins job count.
Their efforts may sometimes be unseen, but they are always felt throughout the project.
```

gounthar commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,176 @@
+[[security]]
+image:/images/post-images/2025/01/08/security.png[Security Update]
+
+At the beginning of 2024, link:https://www.jenkins.io/blog/2024/01/25/sonar-vulnerability-report/[Sonar worked with the Jenkins security team] to address a previously unnoticed vulnerability.
+The issue was very niche and would only execute given a specific set of circumstances and permissions, so there was no larger threat here.
+However, the fact that this vulnerability existed was enough reason for Jenkins security and Sonar to collaborate and resolve this issue.
+Thanks to their in-depth work and tenacity in removing the vulnerability, Jenkins continues to be a secure and well-maintained DevOps platform.
```
```suggestion
At the beginning of 2024, link:https://www.jenkins.io/blog/2024/01/25/sonar-vulnerability-report/[Sonar worked with the Jenkins security team] to address a previously unnoticed vulnerability.
The issue was very niche and would only execute given a specific set of circumstances and permissions, so there was no larger threat here.
However, the existence of this vulnerability was enough reason for Jenkins security and Sonar to collaborate and resolve this issue.
Thanks to their in-depth work and tenacity in removing the vulnerability, Jenkins continues to be a secure and well-maintained DevOps platform.
```

gounthar commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,176 @@
+Over the summer, the Jenkins weekly release line made the move to link:https://www.jenkins.io/blog/2024/06/11/require-java-17/[require Java 17] as a baseline.
+This was an important update to keep up with current usage and start the process of dropping support for Java 11.
+Thanks to the hard work and dedication of Jenkins developers, the transition was successful and we saw a steady rise in JDK 17 installation and usage.
+By the end of October LTS release, the Java 17 requirement had been thoroughly tested and used over the preceding months, ensuring that its inclusion would be safe.
+Since the October release of LTS 2.479.1, we have not seen any major issues with Java 17 usage.
+While there have been some hiccups when it comes to upgrading plugins alongside the Jenkins LTS, these have been addressed as they come up and typically can be resolved by ensuring the plugin is updated before and after upgrading Jenkins.
```
```suggestion
By the end of the October LTS release, the Java 17 requirement had been thoroughly tested and used over the preceding months, ensuring that its inclusion would be safe.
Since the October release of LTS 2.479.1, we have not seen any major issues with Java 17 usage.
While there have been some hiccups when it comes to upgrading plugins alongside the Jenkins LTS, these have been addressed as they come up and typically can be resolved by ensuring the plugin is updated before and after upgrading Jenkins.
```

gounthar commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,176 @@
+
+We've completed one full year of link:https://contributors.jenkins.io/[Contributor Spotlight] posts, highlighting the different backgrounds, skills, and work that make up the Jenkins community.
+This space has allowed us to show appreciation for those who keep Jenkins fresh, strong, and viable so consistently.
+From long-time contributors and early adopters of Jenkins, to recent Google Summer of Code participants, there are all kinds of stories to tell and efforts to acknowledge.
+Thanks to all the contributors who have collaborated with us for the contributor spotlight and for allowing us to share their stories.
+Additionally, new contributors have been welcomed to Jenkins through community projects, including Hacktoberfest and another successful Google Summer of Code.
```
```suggestion
From long-time contributors and early adopters of Jenkins to recent Google Summer of Code participants, there are all kinds of stories to tell and efforts to acknowledge.
Thanks to all the contributors who have collaborated with us for the contributor spotlight and for allowing us to share their stories.
Additionally, new contributors have been welcomed to Jenkins through community projects, including Hacktoberfest and another successful Google Summer of Code.
```

gounthar commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,176 @@
+
+Throughout 2024, there were several events that Jenkins participated in that allowed us to connect with the Jenkins and extended open-source communities.
+We started the year by gathering in Brussels for FOSDEM and a hugely successful link:https://www.jenkins.io/blog/2024/02/28/jenkins-contributor-summit-and-fosdem-recap/[Jenkins contributor summit].
+The contributor summit provided a space for Jenkins contributors to meet and connect in person, get updates and roadmap plans from Jenkins officers, and work on a handful of different projects together.
+It is not often that we are all able to meet in-person, so these kind of events are hugely invaluable.
```
```suggestion
The contributor summit provided a space for Jenkins contributors to meet and connect in person, get updates and roadmap plans from Jenkins officers, and work on a handful of different projects together.
It is not often that we are all able to meet in-person, so these kinds of events are hugely invaluable.
```

gounthar replied:
Not sure at all about this one... 🤔

gounthar reviewed (approved):
Thanks a lot for the recap, Kevin! 👍

janfaracik commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,176 @@
+This allowed us to showcase the work and effort that goes into refining the Jenkins experience even further.
+Thanks to author:janfaracik[Jan Faracik] and author:timja[Tim Jacomb] for delivering this talk and providing so many great updates to the Jenkins UI/UX.
+
+Additional thanks to all those who help enhance the Jenkins UX/UI through their efforts to provide the best experience possible.
```
![bento](https://github.com/user-attachments/assets/93bd73f2-5e16-479f-b459-96d39980514c)

How is this for an image giving an overview of the UX changes?

kmartens27 commented:
Thanks very much again @timja and @janfaracik, I've incorporated the feedback you've provided and added the image you shared earlier. Please let me know if what I've written lines up with the work you've done and illustrates that properly.

kmartens27 commented:
Updated deploy preview: https://deploy-preview-7761--jenkins-io-site-pr.netlify.app/blog/2025/01/08/jenkins-2024-recap/

janfaracik commented:
> Thanks very much again @timja and @janfaracik, I've incorporated the feedback you've provided and added the image you shared earlier. Please let me know if what I've written lines up with the work you've done and illustrates that properly.

Looks great, thanks!

MargoBGH commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,191 @@
+
+image:/images/post-images/2025/01/08/2024-recap.png[Jenkins 2024 recap]
+
+As we start 2025, we wanted to look back at the previous year and highlight some of our favorite moments, biggest projects, and important changes over the last 366 days.
```
```suggestion
As we welcome 2025, we take a moment to reflect on the past year and celebrate some of our most memorable achievements, impactful projects, and significant changes over the last 366 days.
```

MargoBGH commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,191 @@
+image:/images/post-images/2025/01/08/2024-recap.png[Jenkins 2024 recap]
+
+As we start 2025, we wanted to look back at the previous year and highlight some of our favorite moments, biggest projects, and important changes over the last 366 days.
+Regardless of what form a contribution takes, the Jenkins project consists of all of them and without them, we would not be what we are today.
```
```suggestion
Every contribution, no matter its form, plays a vital role in shaping the Jenkins project. Without the collective efforts of our contributors, supporters, and community members, Jenkins wouldn’t be what it is today.
```

MargoBGH commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,191 @@
+
+As we start 2025, we wanted to look back at the previous year and highlight some of our favorite moments, biggest projects, and important changes over the last 366 days.
+Regardless of what form a contribution takes, the Jenkins project consists of all of them and without them, we would not be what we are today.
+From every pull request to the grants and donations that help Jenkins keep up with the needs of the Jenkins community, we offer our deepest thanks and appreciation.
```
```suggestion
From every pull request to the grants and donations that enable Jenkins to meet the evolving needs of our community, we extend our heartfelt gratitude and deepest appreciation. Thank you for being part of our journey!
```

MargoBGH commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,191 @@
+
+Thanks to the efforts of the Jenkins infrastructure team, the Jenkins update center has been migrated to a new mirror-based architecture, which is 10x more cost-effective and is link:https://www.jenkins.io/blog/2024/11/16/new-update-center/[now in production].
+Over several months, Jenkins Infra performed multiple brownouts to test the services and ensure that the migration would be smooth.
+The tests were successful and they were able to detect issues that would have caused major headaches for users.
```
```suggestion
The tests were successful, and they were able to detect issues that would have caused major headaches for users.
```

MargoBGH commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,191 @@
+Thanks to the efforts of the Jenkins infrastructure team, the Jenkins update center has been migrated to a new mirror-based architecture, which is 10x more cost-effective and is link:https://www.jenkins.io/blog/2024/11/16/new-update-center/[now in production].
+Over several months, Jenkins Infra performed multiple brownouts to test the services and ensure that the migration would be smooth.
+The tests were successful and they were able to detect issues that would have caused major headaches for users.
+Thankfully, these issues were resolved and while there were a few bumps post-migration, the update center has been doing very well over the last few weeks.
```
```suggestion
Thankfully, these issues were resolved, and while there were a few bumps post-migration, the update center has been doing very well over the last few weeks.
```

MargoBGH commented on content/blog/2025/01/08/2025-01-08-jenkins-2024-recap.adoc:
```diff
@@ -0,0 +1,191 @@
+Throughout 2024, there were several events that Jenkins participated in that allowed us to connect with the Jenkins and extended open-source communities.
+We started the year by gathering in Brussels for FOSDEM and a hugely successful link:https://www.jenkins.io/blog/2024/02/28/jenkins-contributor-summit-and-fosdem-recap/[Jenkins contributor summit].
+The contributor summit provided a space for Jenkins contributors to meet and connect in person, get updates and roadmap plans from Jenkins officers, and work on a handful of different projects together.
+It is not often that we are all able to meet in-person, so these events are hugely invaluable.
```
```suggestion
It is not often that we are all able to meet in person, so these events are hugely invaluable.
```

MargoBGH reviewed (approved):
This looks great to me! I just made some suggestions to the intro para, but they are just suggestions. :)

kmartens27 commented:
this should not be merged until after the next governance meeting (January 13) and the board has approved the sponsor list/image/sorting

kmartens27 commented:
latest preview link with DevOps Dozen award https://deploy-preview-7761--jenkins-io-site-pr.netlify.app/blog/2025/01/16/jenkins-2024-recap/
